src/ParallelAlgorithms/Interpolation/Callbacks/ObserveSurfaceData.hpp

```diff
@@ -0,0 +1,125 @@
+  static constexpr double fill_invalid_points_with =
+      std::numeric_limits<double>::quiet_NaN();
+
+  template <typename DbTags, typename Metavariables, typename TemporalId>
+  static void apply(const db::DataBox<DbTags>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const TemporalId& temporal_id) {
+    const Strahlkorper<Frame::Inertial>& strahlkorper =
+        get<StrahlkorperTags::Strahlkorper<Frame::Inertial>>(box);
+    const YlmSpherepack& ylm = strahlkorper.ylm_spherepack();
+    const std::array<DataVector, 2>& theta_phi = ylm.theta_phi_points();
+    const DataVector& theta = theta_phi[0];
+    const DataVector& phi = theta_phi[1];
+    const DataVector& sin_theta = sin(theta);
+    const DataVector& radius = ylm.spec_to_phys(strahlkorper.coefficients());
+
+    // Output the inertial-frame coordinates.
+    const std::string& surface_name =
+        pretty_type::short_name<InterpolationTargetTag>();
+    std::vector<TensorComponent> tensor_components{
+        {surface_name + "/InertialCoordinates_x"s,
+         radius * sin_theta * cos(phi)},
+        {surface_name + "/InertialCoordinates_y"s,
+         radius * sin_theta * sin(phi)},
+        {surface_name + "/InertialCoordinates_z"s, radius * cos(theta)}};
+
+    // Output each tag if it is a scalar. Otherwise, throw a compile-time error.
+    // This could be generalized to loop over components, outputting them one at
+    // a time.
+    tmpl::for_each<TagsToObserve>([&box, &surface_name,
+                                   &tensor_components](auto tag_v) {
+      using Tag = tmpl::type_from<decltype(tag_v)>;
+      static_assert(std::is_same_v<typename Tag::type, Scalar<DataVector>>,
+                    "Each tag in TagsToObserve must be a Scalar<DataVector>.");
+      tensor_components.push_back(
+          {surface_name + "/"s + pretty_type::short_name<Tag>(),
+           get(get<Tag>(box))});
+    });
+
+    const std::string& subfile_path{std::string{"/"} + surface_name};
+    const std::vector<size_t> extents_vector{
+        {ylm.physical_extents()[0], ylm.physical_extents()[1]}};
+    const std::vector<Spectral::Basis> bases_vector{
+        2, Spectral::Basis::SphericalHarmonic};
+    const std::vector<Spectral::Quadrature> quadratures_vector{
+        {Spectral::Quadrature::Gauss, Spectral::Quadrature::Equiangular}};
+    const observers::ObservationId& observation_id = observers::ObservationId(
+        InterpolationTarget_detail::get_temporal_id_value(temporal_id),
+        subfile_path + ".vol");
+
+    auto& proxy = Parallel::get_parallel_component<
+        observers::ObserverWriter<Metavariables>>(cache);
+
+    // We call this on proxy[0] because the 0th element of a NodeGroup is
+    // always guaranteed to be present.
+    Parallel::threaded_action<observers::ThreadedActions::WriteVolumeData>(
+        proxy[0], "Surfaces"s, std::string{"/" + surface_name}, observation_id,
```

Comment:
Is it possible to add the name of the 2D volume data file as a const_global_cache_tag like the name of the reduction file and 3D volume files and then just get it out of the cache? I'm wary of hard-coding the name here. Also, you don't document what the name of this file is anywhere.